src/assets/ocisWrapper/ocisWrapper.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Automated Black-Box Tests for System Settings of a Microservice Application
+authorName: Artur Neumann
+authorAvatar: https://www.jankaritech.com/images/2019/06/11/p1070364-c-light-800.jpg
+authorLink: https://www.linkedin.com/in/artur-neumann/
+createdAt: May 22, 2023
+tags: go, API, testing, microservice
+banner: https://github.com/JankariTech/blog/blob/master/src/assets/ocisWrapper/banner.png
+---
+oCIS is the new generation open source file-sync and sharing platform, built by [ownCloud GmbH](https://www.owncloud.com), who also created the ownCloud 10 server based on the LAMP stack. The team of [JankariTech](https://www.jankaritech.com) is part of that development and mainly writes and maintains the automated tests.
+
+## The Challenge 🚀
+
+Because a lot of APIs of oCIS still have to be compatible with ownCloud 10, we have used the existing API tests and (after some adjustment) ran them on oCIS to ensure the backwards compatibility. I have written about that process in [BDD on Software Rewrite](https://blog.jankaritech.com/#/blog/Behaviour%20Driven%20Development/BDD%20on%20Software%20Rewrite).
+
+This worked well for a lot of test-cases, and we could quickly reach a good coverage. But there were some cases where the old tests could not be reused. One of those cases is specific system-wide settings. In ownCloud 10 those settings are often set by the UI, an API call or the designated CLI tool. Also, the changes can be done during runtime and are immediately effective. This is great for an API test: change the setting => check the expected behaviour! Easy!
+
+Because of the different architecture of oCIS, system-wide settings are set through YML files or [environment variables](https://doc.owncloud.com/ocis/next/deployment/services/env-var-note.html). This is great if you run the service in a container environment, but it makes automated testing harder. Logically, the settings are only read once during the start of the service and for any change to take effect, the service needs to be restarted. How to do that as an external black-box API test?
+Not to test those settings or to rely only on manual tests was never an option.
+
+## First Iteration 😵‍💫
+
+We first started with creating different pipelines in CI that start oCIS with defined settings and ran specific tests against that instance. That gave quick results, but soon became confusing because every combination of settings needs its own CI pipeline and a designated set of tests. Additionally, that system makes it hard for developers to run the tests in their environments.
+
```

Comment:
```suggestion
## First Iteration 😵‍💫

We first started by creating different pipelines in CI that start oCIS with defined settings and ran specific tests against that instance. That gave quick results but soon became confusing because every combination of settings needs its own CI pipeline and a designated set of tests. Additionally, that system makes it hard for developers to run the tests in their environments.

```